File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimumDifference = function (arr, k) {
    let n = arr.length;
    let diff = 1000000;
    arr.sort(function(a,b){return a-b});
    for(let i = 0; i <= n - k; i++) {
            let tm = arr[i + k - 1] - arr[i];
        diff = Math.min(tm, diff);
    }
    return diff;
};

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] arr, int k) {

        int n = arr.length;

        int diff = 1000000;
        Arrays.sort(arr);
        for (int i = 0; i <= n - k; i++) {
            int tm = arr[i + k - 1] - arr[i];
            diff = Math.min(tm, diff);
        }
        return diff;
    }
}